### PR TITLE
Update to Substrate v2.0.0

### PR DESCRIPTION
**Pull request by LaurentTrk**
There has been some changes between the Substrate 2.0.0-rc6 and the final one, that need to be reported in this rpc client:
* Use metadata V12
* Change RefCount Account type (https://github.com/paritytech/substrate/pull/7164)

I hope I've done things right ;)

### CODE REVIEW

**Comment by LaurentTrk** [comment]
Need to work on travis tests...

**Review by mikiquantum — APPROVED** [review]
@LaurentTrk LGTM! Thanks for the contribution. Only one comment about removing support for v1 in the Makefile. I will merge up after that.

File: Makefile [hunk 1]
```diff
@@ -47,10 +47,10 @@ test-e2e-deployed: 		## runs only end-to-end (e2e) tests against a deployed test
 	@docker run --rm -e RPC_URL -e TEST_PRIV_KEY gsrpc-test go test -v github.com/centrifuge/go-substrate-rpc-client/teste2e
 
 run-substrate-docker: 		## runs the Substrate 1.0 Default Docker image, this can be used to run the tests
```
[the remark this answers is not in the document]
Comment (LaurentTrk) [reply Makefile]: Done.

[the remark this answers is not in the document]
Comment (LaurentTrk) [reply Makefile]: @mikiquantum would you mind label this PR with hacktoberfest-accepted please ? That's could be great :)

**Comment by mikiquantum** [comment]
@LaurentTrk that is very interesting, I can label the PR, Do you have any more information that I can link as well to the PR && issue?

**Comment by LaurentTrk** [comment]
No more information, the label is enough. Thanks !!

**Pull request merged**